app/Listeners/QueueMessage.php

```diff
@@ -31,18 +31,22 @@ public function __construct(Mailer $mail)
     public function handle(QueueMessageRequest $event)
     {
         $email = $event->email;
-        $type = $email->message->type;
 
-        $this->mail->send('messages.' . $type, ['email' => $email], function ($msg) use ($email) {
-            $content = Email::prepareMessage($email->message, $email->competition);
-
-            // @TODO - Pull from name from contest setting.
-            $msg->from($email->sender, 'Beyonce');
-
-            // @TODO - send to users in competition that triggered the send.
-            // can be an array of email addresses.
-            // this is just sending as a test to the person who made the contest.
-            $msg->to($email->sender, 'shae')->subject($email->message->subject);
-        });
+        foreach ($email->allMessages as $message) {
+            $content = $message['message'];
+            $user = $message['user'];
+            $type = $content->type;
+
+            if ($user->email) {
+                $this->mail->queue('messages.' . $type, ['body' => $content->body], function ($msg) use ($email, $content, $user) {
+                    // Pulled from the contest.
+                    $msg->from($email->contest->sender_email, $email->contest->sender_name);
+
+                    // Send to user.
+                    // CAREFUL! Make sure you are using dev settings or you could actually send an email.
```

Comment:
It might be good to short-circuit this method if `APP_DEBUG === true` just to be safe.